frame/staking-rewards/src/lib.rs

```diff
@@ -599,8 +620,119 @@ pub mod pallet {
 					.try_insert(*asset_id, inflation)
 					.map_err(|_| Error::<T>::ReductionConfigProblem)?;
 			}
+
 			Ok((rewards_btree_map, reductions))
 		}
+
+		// TODO: tests
+		pub(crate) fn reward_acumulation_hook_reward_update_calculation(
+			pool_id: T::RewardPoolId,
+			asset_id: T::AssetId,
+			reward: Reward<T::AssetId, T::Balance>,
+			now: u64,
+		) -> (T::AssetId, Reward<T::AssetId, T::Balance>) {
+			let elapsed_time = match now.safe_sub(&reward.last_updated_timestamp) {
+				Ok(elapsed_time) => elapsed_time,
+				Err(_) => {
+					Self::deposit_event(Event::<T>::RewardAccumulationError { pool_id, asset_id });
+					return (asset_id, reward)
+				},
+			};
+
+			// SAFETY(benluelo): RewardRate::period is non-zero.
+			let periods_surpassed = elapsed_time.div(&reward.reward_rate.period);
+
+			if periods_surpassed.is_zero() {
+				(asset_id, reward)
+			} else {
+				// NOTE(benluelo): used this instead of `safe_mul` since
+				// `T::Balance` can't be multiplied by a `u64` and the definition of
+				// `CheckedMul` constrains the `Rhs` type to `Self`, so `T::balance`
+				// can't `.safe_mul(u64)`
```

Comment:
This reason by it self just not justify the use of `repeat` since we can safely convert u64 to balance given that our runtimes use u128 for it.

Reply:
for this and all related comments: https://github.com/ComposableFi/composable/pull/1359/commits/f45dc1dea2e0ba47ee2ba1facea41583eab6f639